test/GPv2Settlement.test.ts

```diff
@@ -1387,6 +1387,57 @@ describe("GPv2Settlement", () => {
         ),
       ).to.be.revertedWith("fee discount too high");
     });
+
+    it("reverts if order was already partially filled", async () => {
+      const order = prepareOrder({
+        kind: OrderKind.SELL,
+        partiallyFillable: true,
+      });
+      const prices = {
+        [order.sellToken]: order.buyAmount,
+        [order.buyToken]: order.sellAmount,
+      };
+      const orderUid = computeOrderUid(testDomain, order, trader.address);
+
+      const encoder = new SettlementEncoder(testDomain);
+      const executedSellAmount = BigNumber.from(1);
+      await encoder.signEncodeTrade(order, traders[0], SigningScheme.EIP712, {
+        executedAmount: executedSellAmount,
+      });
+
+      await settlement.computeTradeExecutionsTest(
+        encoder.tokens,
+        encoder.clearingPrices(prices),
+        encoder.trades,
+      );
+      expect(await settlement.filledAmount(orderUid)).to.deep.equal(
+        executedSellAmount,
+      );
+
+      const [tokens, trade, transfers, interactions] = await prepareTrade(
+        order,
+      );
+      trade.executedAmount = BigNumber.from(order.sellAmount).sub(
+        executedSellAmount,
+      );
+      expect(trade.executedAmount).to.not.deep.equal(ethers.constants.Zero);
+
+      await authenticator.connect(owner).addSolver(solver.address);
+
+      await expect(
+        settlement.connect(solver).callStatic.computeTradeExecutionsTest(
+          tokens,
+          tokens.map((token) => prices[token]),
+          [trade],
+        ),
+      ).not.to.be.reverted;
+
+      await expect(
+        settlement
+          .connect(solver)
+          .executeSingleTradeTest(tokens, trade, transfers, interactions),
+      ).to.be.revertedWith("order filled");
+    });
```

Comment:
NIce!

Maybe for completeness, there should also be one that checks that it reverts if the order was cancelled on-chain (it doesn't affect coverage, but it would be nice to exist in the test-suite to catch regressions if we change the fast path to allow partially fillable orders for example).